src/components/DatePicker.js

```diff
@@ -1206,37 +1211,36 @@ export default class DatePicker extends Component {
 
     if (isDisabledDay) {
       return (
-        <span
+        <DisabledDay
           key={ `day-${index}` }
-          ref={ dateKey }
           style={ dayStyle }
-          onMouseEnter={ (event) => this._onDayMouseEnter(dateKey, event) }
-          onMouseLeave={ (event) => this._onDayMouseLeave(dateKey, event) }
-          {...this.disabledDayProps}
+          dateKey={ dateKey }
+          onDayMouseEnter={ this._onDayMouseEnter }
+          onDayMouseLeave={ this._onDayMouseLeave }
+          disabledDayProps={ this.disabledDayProps }
         >
           { renderedDay }
-        </span>
+        </DisabledDay>
       );
     }
 
```

Comment:
@nikgraf I do not like the idea of extracting a component which just handles event callbacks to get rid of bindings. A component should be a more logical entity and should be able to handle its styling logic and stuff like making dayKey out of date.

If we can do this kind of component it will help us in not just getting rid of bindings but also simplify these complex components.